components/Wrapper.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Router, { useRouter } from 'next/router';
+
+import Navbar from './Navbar';
+import Header from './Header';
+import DashboardHeader from './DashboardHeader';
+
+const isRegistered = false;
+
+const Wrapper = ({ Component }) => {
+  const { pathname } = useRouter();
+  const isHome = pathname === '/';
+  const isTournament = pathname.substr(0, 13) === '/tournaments/';
+  const isDashboard = pathname.substr(0, 10) === '/dashboard';
+
+  // Handle redirections
+  let redirect = null;
+
+  if (isDashboard && process.env.DASHBOARD_AVAILABLE === 'false') {
+    redirect = '/';
+  }
+  else if (isRegistered && (pathname === '/dashboard' || pathname === '/dashboard/register')) {
+    redirect = '/dashboard/team';
+  }
+  else if (!isRegistered && (pathname === '/dashboard' || pathname === '/dashboard/team')) {
+    redirect = '/dashboard/register';
+  }
+
+  // Redirect to desired path
+  useEffect(() => {
+    redirect && Router.replace(redirect);
+  });
```

Comment:
```suggestion
  },  []);
```
Pour exécuter qu'une fois l'effet